byron/ledger/executable-spec/src/Ledger/Update.hs

```diff
@@ -520,10 +529,14 @@ instance STS UPVOTE where
     , Set (UpId, Core.VKeyGenesis)
     )
   type Signal UPVOTE = Vote
+
+  -- | `S_CfmThdNotReached` is a structural `PredicateFailure`, used to fail
+  -- from one transition rule to the other. The other `PredicateFailure`s are
+  -- all throwable.
   data PredicateFailure UPVOTE
     = ADDVOTEFailure (PredicateFailure ADDVOTE)
     | HigherThanThdAndNotAlreadyConfirmed
```

Comment:
I think this is a structural error as well ...

Reply:
Yes, I think you're right. Double-negation...